cou/utils/juju_utils.py

```diff
@@ -266,6 +313,38 @@ async def connect(self) -> None:
         )
 
     @retry
+    async def get_applications(self) -> dict[str, COUApplication]:
+        """Return list of applications with all relevant information.
+
+        :returns: list of application with all information
+        :rtype: list[COUApplication]
+        """
+        # note(rgildein): We get the applications from the Juju status, since we can get more
+        #                 information the status than from objects. e.g. workload_version for unit
+        full_status = await self.get_status()
+        machines = await self.get_model_machines()
+
```

Comment:
I think you can also get the applications like: `model.applications` and with that access the charm_name and the config without the need of using `self.get_application_config`